src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -1432,7 +1432,12 @@ internal async Task<DebugStore> LoadStore(SessionId sessionId, CancellationToken
                 }
                 else
                 {
-                    await foreach (SourceFile source in context.store.Load(sessionId, loaded_files, token).WithCancellation(token))
+                    var useDebuggerProtocol = false;
+                    (int MajorVersion, int MinorVersion) = await context.SdbAgent.GetVMVersion(token);
+                    if (MajorVersion == 2 && MinorVersion >= 61)
+                        useDebuggerProtocol = true;
+
+                    await foreach (SourceFile source in context.store.Load(sessionId, loaded_files, useDebuggerProtocol ? context.SdbAgent : null, token).WithCancellation(token))
```

Comment:
Can we pass just `context` instead of `loaded_files`, and `sdbagent`? `context`, and a `bool useDebuggerProtocol`.
We'll need to ensure that the `context.LoadedFiles` is set when `GetLoadedFiles` returns. Would that work?